src/app_service/service/contacts/service.nim

```diff
@@ -38,8 +37,8 @@ type
 
 # Local Constants:
 const CheckStatusIntervalInMilliseconds = 5000 # 5 seconds, this is timeout how often do we check for user status.
-const OnlineLimitInSeconds = int(5.5 * 60) # 5.5 minutes
-const IdleLimitInSeconds = int(7 * 60) # 7 minutes
+const FiveMinsOnlineLimitInSeconds = int(5 * 60) # 5 minutes
+const TwoWeeksOnlineLimitInSeconds = int(14 * 24 * 60 * 60) # 2 weeks
```

Comment:
Just wondering why this is on the nim-client side instead of status-go? Ideally, we should just receive signal from status-go and update appropriate contact with the new online status.

Reply:
It seems like a good idea.
Now, we are receiving signal from status-go every time the other peer broadcasts its status.
We keep all updated statuses and times of change in the cache and every 5 seconds we check the cache when the last status for every peer changed. Probably such a cache could be placed in status-go.
FYI @richard-ramos

Reply:
> The 5s caching approach doesn't explain why nim client needs to check these intervals (5min and 2weeks) :thinking: Doesn't a status update come with online status already?

The only explanation I can think of is that this update is emitted only when peers broadcast their status over the network, which means we must check the interval to determine if they are offline or not. Nevertheless, I believe it should be done on status-go side, clients should only be interested in status updates, not the logic of how to obtain them. It also explains the "5s loop cache" mechanism, which is not really the optimization solution, it is rather more checking in the loop if contact has become offline or not.

Reply:
@osmaczko this is a good question. Regarding members/users the client is only interested in 2 states Online/Inactive. And I think the whole logic of checking statuses of members could be moved to status-go. At least, I do not see any problems with that. I will talk to Parvesh from mobile about a possibility of doing this.